corehq/apps/userreports/sql/adapter.py

```diff
@@ -177,6 +177,12 @@ def save_rows(self, rows, use_shard_col=True):
             for query in queries:
                 session.execute(query)
 
+        register_data_source_row_change(
+            domain=self.config.domain,
+            data_source_id=self.config._id,
+            row_changes=[{key: str(value) for key, value in row.items()} for row in formatted_rows],
```

Comment:
I think this could just use `doc_ids` variable defined above

Reply:
Ha-ha, good catch!

Reply:
https://github.com/dimagi/commcare-hq/pull/33877/commits/395a4627d252ca929a3233135753333ea55b8771